src/adaptation/ecap/XactionRep.cc

```diff
@@ -246,7 +246,7 @@ Adaptation::Ecap::XactionRep::start()
         // retrying=false because ecap never retries transactions
         adaptHistoryId = ah->recordXactStart(service().cfg().key, current_time, false);
         SBuf matched;
-        for (auto h: Adaptation::Config::metaHeaders) {
+        for (const auto &h: Adaptation::Config::metaHeaders()) {
```

Comment:
Using a reference here (and similar loops modified in this PR) may reduce overheads associated iterating smart pointers. The type change mimics what was done to similar iteration code in master/v7 commit https://github.com/squid-cache/squid/commit/984577acdcd448a8a6893e04070712cf27d58a3b that this PR is fixing. If it works, I think this type change is an in-scope improvement.